cmd/uplink/cmd/ls.go

```diff
@@ -26,7 +27,8 @@ func init() {
 		Short: "List objects and prefixes or all buckets",
 		RunE:  list,
 	}, RootCmd)
-	recursiveFlag = lsCmd.Flags().Bool("recursive", false, "if true, list recursively")
+	lsRecursiveFlag = lsCmd.Flags().Bool("recursive", false, "if true, list recursively")
```

Comment:
so if I call ls with recursive and I don't have the encryption key, I won't be able to see my files? But If I call it with encrypted, I will be able to see them, but encoded? What if I call encrypted with the right encryption key?

Reply:
You will get all `url-base64-encoded(encrypted)` paths when using `--encrypted`. No path decryption is attempted.

Reply:
From a usability standpoint it might be nice if we always try to decrypt but fail-over to url-base64-encoding. When there is an object in a bucket that was encrypted with a different key it can be difficult to determine which object that is just by using `uplink ls --encrypted`. Currently you won't be able to `uplink ls` a prefix that contains a path component that was encrypted with that other key.

Reply:
Automatic fail-over with tools that are used in scripts can be problematic.

Reply:
Maybe instead a subcommand to convert between unencrypted and encrypted-encoded paths or something then.

Reply:
Or we could add a flag to enable that behavior.